ledger/src/blockstore_meta.rs

```diff
@@ -112,6 +116,33 @@ pub struct Index {
     coding: ShredIndex,
 }
 
+#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq, Eq)]
+pub struct IndexV2 {
+    pub slot: Slot,
+    data: ShredIndexV2,
+    coding: ShredIndexV2,
+}
+
+impl From<IndexV2> for Index {
+    fn from(index: IndexV2) -> Self {
+        Index {
+            slot: index.slot,
+            data: index.data.into(),
+            coding: index.coding.into(),
+        }
+    }
+}
+
+impl From<Index> for IndexV2 {
+    fn from(index: Index) -> Self {
+        IndexV2 {
+            slot: index.slot,
+            data: index.data.into(),
+            coding: index.coding.into(),
+        }
+    }
+}
```

Comment:
Not every `Index` infallibly can be converted to an `IndexV2` if we change `MAX_DATA_SHREDS_PER_SLOT`.
The code silently ignoring this in particular makes it possible that given an `index: Index`:
```rust
From<IndexV2>(From<Index>(index)) != index
````
which is asking for bugs and breakage.